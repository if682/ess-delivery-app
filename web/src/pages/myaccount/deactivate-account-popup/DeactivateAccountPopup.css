.buttons-container{
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    gap: 1rem;
    margin-top: 2rem;
}

.main-text{
    width: 397px;
    height: 28px;
    font-style: normal;
    font-weight: 600;
    font-size: 24px;
    line-height: 28px;
    text-align: center;

    color: #272727;

}