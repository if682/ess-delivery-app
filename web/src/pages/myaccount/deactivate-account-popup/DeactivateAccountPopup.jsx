import Modal from 'react-bootstrap/Modal';
import Button from 'react-bootstrap/Button';
import Cookies from 'js-cookie';
import { useNavigate} from 'react-router-dom';
import './DeactivateAccountPopup.css'

export const DeactivateAccountPopup = (props) => {
    const navigate = useNavigate();

    //retorna para a página de login e remove o token de acesso do usuário
    function DeactivateAccount(){
        navigate('/login');
        Cookies.remove("token");
    }


    return (
        <Modal
        {...props}
        size="md"
        aria-labelledby="contained-modal-title-vcenter"
        centered
        >
        <Modal.Header closeButton {...props}>
        </Modal.Header>
        <Modal.Body>
            <text className='main-text'>Deseja mesmo desativar sua conta?</text>
            <br/>
            <div className='buttons-container'>
                <Button className='btn-primary' onClick={() => DeactivateAccount()}>Confirmar</Button>
                <Button className='btn-secondary' onClick={props.onHide}>Cancelar</Button>
            </div>
            
        </Modal.Body>
        </Modal>
    );
}
