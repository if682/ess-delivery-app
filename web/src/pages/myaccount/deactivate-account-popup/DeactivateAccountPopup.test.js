import { render, screen, fireEvent, waitFor } from "@testing-library/react";
import { DeactivateAccountPopup } from "./DeactivateAccountPopup";
import { BrowserRouter } from "react-router-dom";

describe("DeactivateAccountPopup", () => {
    let onHideMock, props;
    
    //mocka a função onHide e cria o props que são recebidos pelo popup
    beforeEach(() => {
        onHideMock = jest.fn();
        props = {
            show: true,
            onHide: onHideMock
        };
    });
    
    //Verifica se o popup é renderizado corretamente
    it('renders correctly', () => {
        const { container } = render(<BrowserRouter><DeactivateAccountPopup {...props} /></BrowserRouter>);
        expect(container.firstChild).toMatchSnapshot();
    });

    //Verifica se a função mockada onHide é chamada quando o botão de cancelar é clicado 
    it('should call onHide when cancel button is clicked', () => {
        const { getByText } = render(<BrowserRouter><DeactivateAccountPopup {...props} /></BrowserRouter>);
        fireEvent.click(getByText('Cancelar'));
        expect(onHideMock).toHaveBeenCalled();
    });

    //Verifica se o site expõe a tela de login após um clique no botão de confirmar
    it('should navigate to the login page when confirm button is clicked', () => {
        const { getByText } = render(<BrowserRouter><DeactivateAccountPopup {...props} /></BrowserRouter>);
        fireEvent.click(getByText('Confirmar'));
        expect(window.location.pathname).toBe('/login');
    });
    
});
