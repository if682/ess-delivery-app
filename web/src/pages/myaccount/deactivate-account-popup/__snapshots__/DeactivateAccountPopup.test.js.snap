// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeactivateAccountPopup renders correctly 1`] = `null`;
